torch/distributed/fsdp/_optim_utils.py

```diff
@@ -1272,6 +1276,436 @@ def _is_named_optimizer(optim_state_dict: Dict[str, Any]) -> bool:
     return isinstance(key, str)
 
 
+@dataclass
+class StateInfo:
```

Comment:
nit: Do you think it is worth documenting that keys are state names, e.g. `"exp_avg"`?